manager/integration/tests/test_node.py

```diff
@@ -2611,3 +2614,80 @@ def test_disk_eviction_with_node_level_soft_anti_affinity_disabled(): # NOQA
     7. Make replica count as 1, Delete the replicas on other 2 nodes.
        Verify the data from the volume.
     """
+    # Step 1
+    node_soft_anti_affinity_setting = \
+        client.by_id_setting(SETTING_REPLICA_NODE_SOFT_ANTI_AFFINITY)
+    client.update(node_soft_anti_affinity_setting, value="false")
+
+    # Step 2
+    nodes = client.list_node()
+    vol_name = common.generate_volume_name()
+    volume = client.create_volume(name=vol_name,
+                                  size=SIZE, numberOfReplicas=len(nodes))
+    common.wait_for_volume_detached(client, vol_name)
+
+    lht_hostId = get_self_host_id()
+    volume.attach(hostId=lht_hostId)
+    volume = common.wait_for_volume_healthy(client, vol_name)
+
+    # Step 3
+    data = common.write_volume_random_data(volume)
+    common.check_volume_data(volume, data)
+
+    # Step 4
+    node = client.by_id_node(lht_hostId)
+    test_disk_path = create_host_disk(client, "vol-test", str(Gi), lht_hostId)
+    test_disk = {"path": test_disk_path, "allowScheduling": True}
+
+    update_disks = get_update_disks(node.disks)
+    update_disks["test-disk"] = test_disk
+    node = update_node_disks(client, node.name, disks=update_disks,
+                             retry=True)
+    node = common.wait_for_disk_update(client, lht_hostId, len(update_disks))
+    assert len(node.disks) == len(update_disks)
+
+    # Step 5
+    for disk in update_disks.values():
+        if disk["path"] != test_disk_path:
+            disk.allowScheduling = False
+            disk.evictionRequested = True
+
+    node = update_node_disks(client, node.name, disks=update_disks, retry=True)
+
+    # Step 6
+    replica_path = test_disk_path + '/replicas'
+    assert os.path.isdir(replica_path)
+
+    for i in range(common.RETRY_COMMAND_COUNT):
+        if len(os.listdir(replica_path)) > 0:
+            break
+        time.sleep(common.RETRY_EXEC_INTERVAL)
+    assert len(os.listdir(replica_path)) > 0
+
+    # Step 7
+    replica_count = 1
+    volume = client.by_id_volume(vol_name)
+    volume = volume.updateReplicaCount(replicaCount=replica_count)
+    volume = common.wait_for_volume_healthy(client, vol_name)
+
+    for r in volume.replicas:
+        if r.hostId != lht_hostId:
+            volume.replicaRemove(name=r.name)
+
+    common.check_volume_data(volume, data)
+
+    # Schedule default disk
+    def finalizer():
+        node = client.by_id_node(lht_hostId)
+        update_disks = get_update_disks(node.disks)
+        for disk in update_disks.values():
+            if disk["path"] == test_disk_path:
+                disk.allowScheduling = False
+            else:
+                disk.allowScheduling = True
+                disk.evictionRequested = False
+
+        update_node_disks(client, node.name, disks=update_disks, retry=True)
+        common.cleanup_all_volumes(client)
```

Comment:
Can we move this to the general clean up? https://github.com/longhorn/longhorn/issues/4066

Reply:
HI @innobead ,
I found we already have `cleanup_test_disks` in common can reuse.
Therefore In finalizer I only kept `cleanup_all_volumes` let no volume mounted to extra disk and let `cleanup_test_disks` can work properly at test case teardown.
Also add enable_default_disk in common teardown for review.